File: backend/app/schemas/shortlist.py
from pydantic import BaseModel, Field


class ShortlistWeights(BaseModel):
    screening: float
    economics: float
    logistics: float
    confidence: float


class ShortlistScoreBreakdown(BaseModel):
    screening_score: float
    economics_score: float
    logistics_score: float
    confidence_score: float
    final_score: float


class ShortlistCandidate(BaseModel):
    shortlist_rank: int
    recommendation: str
    readiness_label: str
    plant_id: str
    scenario_id: str
    scenario_result_id: str
    site_name: str
    unit_name: str
    province: str | None
    city: str | None
    capacity_mw: float | None
    screening_rank: int
    composite_score: float
    co2_tpy: float | None
    captured_co2_tpy: float | None
    methanol_tpy: float | None
    h2_required_tpy: float | None
    electrolyzer_required_mw: float | None
    gross_revenue_usd_per_year: float | None
    estimated_lcom_usd_ton: float | None
    nearest_port_name: str | None
    nearest_port_distance_km: float | None
    nearest_port_readiness_score: float
    port_proximity_score: float
    score_breakdown: ShortlistScoreBreakdown
    data_confidence_label: str
    data_gap_count: int
    key_bottleneck: str
    decision_rationale: list[str] = Field(default_factory=list)
    next_actions: list[str] = Field(default_factory=list)


class ShortlistSummary(BaseModel):
    candidate_count: int
    shortlist_count: int
    top_candidate: str | None
    top_score: float | None
    shortlist_captured_co2_tpy: float
    shortlist_methanol_tpy: float
    shortlist_gross_revenue_usd_per_year: float
    average_shortlist_lcom_usd_ton: float | None


class ShortlistDecisionMatrix(BaseModel):
    scheme: str
    top_n: int
    method: str
    weights: ShortlistWeights
    summary: ShortlistSummary
    candidates: list[ShortlistCandidate]
    warnings: list[str] = Field(default_factory=list)

File: frontend/types/shortlist.ts
export type ShortlistWeights = {
  screening: number;
  economics: number;
  logistics: number;
  confidence: number;
};

export type ShortlistScoreBreakdown = {
  screening_score: number;
  economics_score: number;
  logistics_score: number;
  confidence_score: number;
  final_score: number;
};

export type ShortlistCandidate = {
  shortlist_rank: number;
  recommendation: string;
  readiness_label: string;
  plant_id: string;
  scenario_id: string;
  scenario_result_id: string;
  site_name: string;
  unit_name: string;
  province: string | null;
  city: string | null;
  capacity_mw: number | null;
  screening_rank: number;
  composite_score: number;
  co2_tpy: number | null;
  captured_co2_tpy: number | null;
  methanol_tpy: number | null;
  h2_required_tpy: number | null;
  electrolyzer_required_mw: number | null;
  gross_revenue_usd_per_year: number | null;
  estimated_lcom_usd_ton: number | null;
  nearest_port_name: string | null;
  nearest_port_distance_km: number | null;
  nearest_port_readiness_score: number;
  port_proximity_score: number;
  score_breakdown: ShortlistScoreBreakdown;
  data_confidence_label: string;
  data_gap_count: number;
  key_bottleneck: string;
  decision_rationale: string[];
  next_actions: string[];
};

export type ShortlistDecisionMatrix = {
  scheme: string;
  top_n: number;
  method: string;
  weights: ShortlistWeights;
  summary: {
    candidate_count: number;
    shortlist_count: number;
    top_candidate: string | null;
    top_score: number | null;
    shortlist_captured_co2_tpy: number;
    shortlist_methanol_tpy: number;
    shortlist_gross_revenue_usd_per_year: number;
    average_shortlist_lcom_usd_ton: number | null;
  };
  candidates: ShortlistCandidate[];
  warnings: string[];
};
